python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -1084,22 +1085,17 @@ def get_mediaproducts(self):
 
         :returns: str
         """
-        media_products_path = os.path.join(self._get_repodata_path(), 'media.1/products')
-        try:
-            (s,b,p,q,f,o) = urlparse(self.url)
-            if p[-1] != '/':
-                p = p + '/'
-            p = p + 'media.1/products'
-        except (ValueError, IndexError, KeyError) as e:
-            return None
-        url = urlunparse((s,b,p,q,f,o))
+        url = 'media.1/products'
+        media_products_path = os.path.join(self._get_repodata_path(), url)
+        grabber = urlgrabber.grabber.URLGrabber()
+        mirror_group = MirrorGroup(grabber, self.repo.urls)
         try:
             urlgrabber_opts = {}
             self.set_download_parameters(urlgrabber_opts, url, media_products_path)
-            urlgrabber.urlgrab(url, media_products_path, **urlgrabber_opts)
+            mirror_group.urlgrab(url, media_products_path, **urlgrabber_opts)
         except URLGrabError as exc:
             repl_url = suseLibURL(url).getURL(stripPw=True)
-            if not hasattr(exc, "code"):
+            if not hasattr(exc, "code") and exc.errno != 256:
```

Comment:
What's errno 256? Can we give it a name in the code?

Reply:
This is the "no more mirrors to try" error once we tried all defined mirrors. Agree, let's clarify this by moving this value to a constant.